modules/ml/src/rtrees.cpp

```diff
@@ -235,35 +236,35 @@ class DTreesImplForRTrees : public DTreesImpl
                     oobperm.resize(n_oob);
                     for( i = 0; i < n_oob; i++ )
                         oobperm[i] = oobidx[i];
+                    for (i = n_oob - 1; i > 0; --i)  //Randomly shuffle indices so we can permute features
```

Comment:
it looks like the oobidx[n_oob-1] always stays at the same location and other indices are only exchanged with smaller indices. of course, they have a chance to move "downstream" at later iterations, but still. would it make sense to replace r_i = rng.uniform(0, i+1) with r_i = rng.uniform(0, n_oob) ?